src/pages/default/profiles/Profiles.tsx

```diff
@@ -28,38 +26,11 @@ const ProfilesScreen = ({ navigation }: Props) => {
       const managedIdentities = await IdentityAgentManager.listIdentities();
       const unfilteredRows = await Promise.all(
         managedIdentities.map(async (identity): Promise<Row | undefined> => {
-          const web5 = IdentityAgentManager.web5(identity);
-
-          const queryResult = await web5.dwn.records.query({
-            message: {
-              filter: {
-                protocol: profileProtocol.protocol,
-                protocolPath: "profile",
-              },
-            },
-          });
-
-          // The Records returned with a query result DO NOT have access
-          // to the data. Get the recordId we're interested in and
-          // make an explicit read request to acquire a Record that DOES
-          // have access to the data.
-          const recordId = queryResult.records?.at(0)?.id;
-          if (!recordId) {
+          const profile = await fetchProfile(identity);
+          if (!profile) {
             return undefined;
           }
 
-          const readResult = await web5.dwn.records.read({
-            message: {
-              recordId,
-            },
-          });
-
-          if (!readResult) {
-            return undefined;
-          }
-
-          const profile = (await readResult.record.data.json()) as Profile;
```

Comment:
This profile fetching logic was moved to `src/features/identity/fetch-profile.ts` in this PR, as it's now being used for both the `ProfilesScreen` and `ProfileDetailScreen`